src/pages/home/ReportScreen.js

```diff
@@ -37,6 +38,7 @@ class ReportScreen extends React.Component {
 
         if (reportChanged) {
             this.prepareTransition();
+            this.storeCurrentlyViewedReport();
```

Comment:
This replaces the logic here:

https://github.com/Expensify/Expensify.cash/blob/d3f1d28478357e2a66b155e660def191d6ed79a8/src/libs/Navigation/NavigationRoot.js#L104-L109

The `ReportScreen` renders the last viewed report so we don't need to check per each navigation change, but only save the last report that was viewed in `ReportScreen`

Reply:
This change makes sense to me. I don't see how it's different from what we had before (maybe a bit cleaner). Was there any other motivation for the change or just cleaning stuff up?

Reply:
Primarily clean up, it should still work the same as before. I can revert this you think it might cause some issue down the road